### PR TITLE
Add decorator examples

### PR DESCRIPTION
**Pull request by abraham**
Fixes https://github.com/Polymer/lit-element/issues/803

`customElement` wasn't rendering `@example` for some reason so it's just part of the description.

### CODE REVIEW

**Comment by abraham** [comment]
@justinfagnani ping.

**Comment by abdonrd** [comment]
I have a question about the `@query()` decorator.
What is the good practice to initialize?
Because TS complains if it is not initialized.

Screenshot 2019-11-15 at 15 22 56

> Property 'form' has no initializer and is not definitely assigned in the constructor.

**Comment by abraham** [comment]
Depending on how often you are referencing the property and how confident the element will always be available you could make it `private form!: HTMLFormElement;`. It's somewhat frowned on but this is one of the usecases. Alternaltivly you could use `?` instead but if you use that property a lot you'll always have to gate it's usage with value checks.

File: src/lib/decorators.ts [hunk 1]
```diff
@@ -173,6 +207,24 @@ export function query(selector: string) {
...
  * that executes a querySelectorAll on the element's renderRoot.
  *
+ * @param selector A DOMString containing one or more selectors to match.
+ *
+ * See: https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll
+ *
+ * @example
+ *
+ *     class MyElement {
+ *       @query('div')
```
Comment (justinfagnani) [inline src/lib/decorators.ts]: Needs to be `queryAll`

File: src/lib/decorators.ts [hunk 2]
```diff
@@ -150,6 +166,24 @@ export function property(options?: PropertyDeclaration) {
...
  * executes a querySelector on the element's renderRoot.
  *
+ * @param selector A DOMString containing one or more selectors to match.
+ *
+ * See: https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector
+ *
+ * @example
+ *
+ *     class MyElement {
+ *       @query('.first')
```
Comment (justinfagnani) [inline src/lib/decorators.ts]: Maybe use an id here?

**Pull request merged**